--- a/src/compat/version/linux/version.h
+++ b/src/compat/version/linux/version.h
@@ -5,8 +5,12 @@
 #ifndef COMPAT_VERSION
 #define COMPAT_VERSION 5
 #endif
-#define KERNEL_VERSION(a, b, c) (((a) << 24) + ((b) << 16) + (c))
-#define LINUX_VERSION_CODE KERNEL_VERSION(COMPAT_VERSION, COMPAT_PATCHLEVEL, COMPAT_SUBLEVEL)
+
+#ifndef KERNEL_VERSION
+# define KERNEL_VERSION(a, b, c) (((a) << 24) + ((b) << 16) + (c))
+#endif
+
+#ifndef LINUX_VERSION_CODE
+# define LINUX_VERSION_CODE KERNEL_VERSION(COMPAT_VERSION, COMPAT_PATCHLEVEL, COMPAT_SUBLEVEL)
+#endif
